modules/gentoo/2020.lua.core

```diff
@@ -60,9 +60,23 @@ elseif arch == "avx" then
 else
         setenv("PIP_CONFIG_FILE", "/cvmfs/soft.computecanada.ca/config/python/pip-gentoo.conf")
 end
-setenv("MII_INDEX_FILE", "/cvmfs/soft.computecanada.ca/custom/mii/data/" .. arch .. "_" .. vendor_id)
-if os.getenv("RSNT_ENABLE_LMOD_CACHE") == "yes" then
-	append_path("LMOD_RC", "/cvmfs/soft.computecanada.ca/config/lmod/lmodrc/cache_" .. arch .. "_" .. vendor_id .. ".lua")
+setenv("MII_INDEX_FILE", "/cvmfs/soft.computecanada.ca/custom/mii/data/" .. arch .. "_" .. cpu_vendor_id)
+local user = os.getenv("USER","unknown")
+local home = os.getenv("HOME",pathJoin("/home",user))
+-- this logic is not needed in spider, avail or other modes
+if mode() == "load" or mode() == "unload" then
+	if os.getenv("RSNT_ENABLE_LMOD_CACHE") == "yes" then
+		append_path("LMOD_RC", "/cvmfs/soft.computecanada.ca/config/lmod/lmodrc/cache_" .. arch .. "_" .. cpu_vendor_id .. ".lua")
+	elseif os.getenv("RSNT_ENABLE_LMOD_CACHE") == "auto" then
+		local has_local_paths = os.getenv("RSNT_LOCAL_MODULEPATHS") or "no"
```

Comment:
this could be much simplified, and also take advantage of short-circuit boolean logic:

```
if not os.getenv("RSNT_LOCAL_MODULEPATHS") and not posix.stat(pathJoin(home, ".local/easybuild/modules")) then
	append_path("LMOD_RC", "/cvmfs/soft.computecanada.ca/config/lmod/lmodrc/cache_" .. arch .. "_" .. cpu_vendor_id .. ".lua")
end

Reply:
done